functions.cmake

```diff
@@ -235,6 +235,10 @@ function(_add_executable)
     if(NOT WIN32)
       add_executable(${ARGV} ${VERSION_FILE_PATH})
       disable_pax_mprotect(${ARGV})
+      if (HAVE_LTO AND CLR_CMAKE_BUILD_COREBUNDLE)
```

Comment:
I am not sure I understand why we are adding the options to targets here instead of setting them globally like any other compiler options. Do we need to exclude some sources from being compiled with the -flto?

Reply:
There were a couple projects that don't go through this function and when those weren't compiled with `-flto`, the resulting binary was ~50KB smaller (with the startup times about the same). I put this change up to check that it would work on our different configurations of CI machines while I was still looking at which projects those were exactly that was resulting in the difference.

Having looked at it more, it is compiling just the `coreclrpal` project with `-flto` that is resulting in the difference, so I could set the options globally and have that one project remove it.